src/tools/kata-ctl/src/utils.rs

```diff
@@ -38,6 +40,120 @@ pub fn get_kernel_version(proc_version_file: &str) -> Result<String> {
     Ok(kernel_version)
 }
 
+const OS_RELEASE: &str = "/etc/os-release";
+
+//Clear Linux has a different path (for stateless support)
+const OS_RELEASE_CLR: &str = "/usr/lib/os-release";
+
+const UNKNOWN: &str = "unknown";
+
+pub fn get_distro_details(os_release: &str, os_release_clr: &str) -> Result<(String, String)> {
+    let files = vec![os_release, os_release_clr];
+    let mut name = String::new();
+    let mut version = String::new();
+
+    for release_file in files.iter() {
+        match fs::read_to_string(release_file) {
+            Err(e) => {
+                if e.kind() == std::io::ErrorKind::NotFound {
+                    continue;
+                } else {
+                    return Err(anyhow!(
+                        "Error reading file {}: {}",
+                        release_file,
+                        e.to_string()
+                    ));
+                }
+            }
+            Ok(contents) => {
+                let lines = contents.lines();
+
+                let get_field = |line: &str| -> Result<String> {
+                    let fields: Vec<&str> = line.split("=").collect();
+                    if fields.len() < 2 {
+                        return Err(anyhow!("Unexpected file contents for {}", release_file));
+                    } else {
+                        let val = fields[1].trim();
+                        return Ok(String::from(val));
+                    }
+                };
+
+                for line in lines {
+                    if line.starts_with("NAME=") && name.is_empty() {
+                        name = get_field(line)?;
+                    } else if line.starts_with("VERSION_ID=") && version.is_empty() {
+                        version = get_field(line)?;
+                    }
+                }
+                if !name.is_empty() && !version.is_empty() {
+                    return Ok((name, version));
+                }
+            }
+        }
+    }
+
+    if name.is_empty() {
+        name = String::from(UNKNOWN);
+    }
+
+    if version.is_empty() {
+        version = String::from(UNKNOWN);
+    }
+
+    Ok((name, version))
+}
+
+pub fn get_generic_cpu_details(cpu_info_file: &str) -> Result<(String, String)> {
+    let cpu_info = get_single_cpu_info(cpu_info_file, "\n\n")?;
+    let lines = cpu_info.lines();
+    println!("Single cpu info: {}", cpu_info);
+    let mut vendor = String::new();
+    let mut model = String::new();
+
+    let get_field = |line: &str| -> Result<String> {
```

Comment:
There are wo `get_field` closuers in two different functions, this can be defined as a fn like `get_field(line: &str, delimiter: &str) -> Result<String>`

Reply:
Done. I have converted the closures into a single function.